atlp-rwanda/atlp-pulse-fn
# integrated BE with FE on user registration and user invite

# PR Description
This PR implemented the user registration functionality and addressed requested changes on managing students in a cohort.
# Description of tasks that were expected to be completed

- [x] user registration
- [x] invitation to register
- [x]  email once trainee is added to cohort
- [x] dynamically managing students as opposed to the previous that was accessed by a coordinator
- [x] drop down of users while adding trainee in a cohort instead of typing their email

# How has this been tested?

-    firstly head to this backend PR [link](https://github.com/atlp-rwanda/atlp-pulse-bn/pull/34) and follow the instructions. 
-    git clone 
-    git checkout registering-user-integration
-    npm install  &&  npm run dev 
-  open a new tab in your web browser paste : http://localhost:3000/
N.B  make sure the backend branch is also running

   Below is a list to test the backend PR it will be connected to

-    Head to the sign in button and click on  sign up , fill in the form with a real email address as it will be sent an email.
-  Now head to the sign-in button again and login organization 1 as organization, login a coordinator with the credentials: email:coordinator@pulse.com
password: Andela123
then head to the trainee section on the side bar, click the add button, you will see the email of the new created user in the drop down, select it and choose the cohort in the drop down, now click save, you will see the new added trainee reflected in the table.
- Check in the inbox of the email you've used to create an account

  Repeat the same process and login another user using organization 1 with credentials 
  email:admin@pulse.com
  password: Andela123
  
  Now head to the trainee section on the sidebar, you will find that there are now two buttons add and invite .
  register another user and follow the same process followed by the coordinator  to add them in the cohort.
  You can also invite a user by clicking on the invite button, type a valid email click invite then check its inbox
  

# Number of Commits
1
# Screenshots (If appropriate)
# Please check this Checklist before you submit your PR:
